### PR TITLE
#5505 - Create Community Stake Banner

### PR DESCRIPTION
**Pull request by Miaplacidus**
This PR completes work detailed in issue #5505, which asks for the creation of a new community stake banner (designs [here](https://www.figma.com/file/KOEVNGXStBhmX3NnYq7MxP/Community-Stake?type=design&node-id=1117-201428&mode=design&t=EY6CaVkOdud2cKZD-0)). 

Ignore the missing URLs for 'Buy Stake' or 'Learn More'. That behavior will be implemented later. 

Note that this work is blocked by issue #5560. 

## Link to Issue
Closes: #5505

## Description of Changes
- creates a new banner (based on the `CWBanner` component) with a button for buying stake in a community, another for learning more about buying stake, and a footer.

## Test Plan
- visit the `/components` page and view the community stake banner

## Other Considerations
- blocked by work in PR #5879 (issue #5560)

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWCommunityStakeBanner/CWCommunityStakeBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            navigate('/');
+          },
+          iconRight: 'externalLink',
+        },
+      ]}
+      onClose={onClose}
+      footer={
+        <>
+          <CWText type="caption">
+            Buying a stake will automatically place you in group
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWCommunityStakeBanner/CWCommunityStakeBanner.tsx:47]: I think the colour for this should be gray, not black

File: packages/commonwealth/client/styles/components/component_kit/cw_component_showcase.scss [hunk 2]
```diff
@@ -200,7 +200,7 @@ $width: 360px;
       gap: 8px;
       margin-bottom: 15px;
 
-      > .Text {
+      >.Text {
```
Comment (masvelio) [inline packages/commonwealth/client/styles/components/component_kit/cw_component_showcase.scss:203]: again, this formatting does not seem to be right

File: packages/commonwealth/client/styles/components/component_kit/new_designs/CWTag.scss [hunk 3]
```diff
@@ -172,7 +173,7 @@
       margin-left: 8px;
 
       &:hover {
-        > svg {
+        >svg {
```
Comment (masvelio) [inline packages/commonwealth/client/styles/components/component_kit/new_designs/CWTag.scss:176]: same here

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWCommunityStakeBanner/CWCommunityStakeBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import React from 'react';
+import CWBanner from 'views/components/component_kit/new_designs/CWBanner';
+import { CWText } from '../../cw_text';
+import { CWTag } from '../CWTag';
+
+type CWCommunityStakeBannerProps = {
+  onClose: () => void;
+  groupName: string;
+};
+export const CWCommunityStakeBanner = ({
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWCommunityStakeBanner/CWCommunityStakeBanner.tsx:11]: My opinion is that `CWComponents` should be design-system components that are building blocks for other components, eg `CWBanner`. CommunityStakeBanner is a specific implementation on top of CWBanner, thus, my proposition is that we should:
1. name the file & component `CommunityStakeBanner`
2. move it to `views/components/CommunityStakeBanner` directory

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWCommunityStakeBanner/CWCommunityStakeBanner.tsx (hunk 1 above)
Comment (Miaplacidus) [reply packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWCommunityStakeBanner/CWCommunityStakeBanner.tsx:47]: @masvelio 
Thanks for the review!
Updated the color to #656167. Nice catch!

File: packages/commonwealth/client/styles/components/component_kit/cw_component_showcase.scss (hunk 2 above)
Comment (Miaplacidus) [reply packages/commonwealth/client/styles/components/component_kit/cw_component_showcase.scss:203]: Updated my editor so this should stop happening. Thanks!

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWCommunityStakeBanner/CWCommunityStakeBanner.tsx (hunk 4 above)
Comment (Miaplacidus) [reply packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWCommunityStakeBanner/CWCommunityStakeBanner.tsx:11]: Agreed and updated accordingly!

**Review by masvelio — APPROVED** [review]
💪

**Review by Israellund — APPROVED** [review]
LGTM

File: packages/commonwealth/client/styles/components/component_kit/new_designs/CWBanner.scss [hunk 5]
```diff
@@ -75,6 +75,16 @@
     flex-wrap: wrap;
   }
 
+  .footer {
+    display: flex;
+    margin-top: 16px;
+    gap: 4px;
+
+    .description {
+      color: #656167;
```
[the remark this answers is not in the document]
Comment (Miaplacidus) [reply packages/commonwealth/client/styles/components/component_kit/new_designs/CWBanner.scss:84]: Updated to color palette variable!

**Pull request merged**